smartcontract/service/native/ontid/utils.go

```diff
@@ -45,26 +43,34 @@ func checkIDExistence(srvc *native.NativeService, encID []byte) bool {
 }
 
 const (
-	FIELD_PK byte = 1 + iota
-	FIELD_ATTR
-	FIELD_RECOVERY
+	flag_exist = 0x01
+
+	FIELD_VERSION byte = 0
+	FLAG_VERSION  byte = 0x01
+
+	FIELD_PK       byte = 1
+	FIELD_ATTR     byte = 2
+	FIELD_RECOVERY byte = 3
 )
 
 func encodeID(id []byte) ([]byte, error) {
 	length := len(id)
 	if length == 0 || length > 255 {
 		return nil, errors.New("encode ONT ID error: invalid ID length")
 	}
-	enc := []byte{byte(length)}
+	//enc := []byte{byte(length)}
+	enc := append(utils.OntIDContractAddress[:], byte(length))
 	enc = append(enc, id...)
 	return enc, nil
 }
 
 func decodeID(data []byte) ([]byte, error) {
-	if len(data) == 0 || len(data) != int(data[0])+1 {
+	prefix := len(utils.OntIDContractAddress)
+	size := len(data)
+	if size == 0 || size < prefix || size != int(data[prefix])+1+prefix {
```

Comment:
`size != int(data[prefix])+1+prefix` is enough

Reply:
may be out of range